docs/cache_extensibility.md

```diff
@@ -0,0 +1,125 @@
+# MSAL Cache Key Extensibility Proposal
+
+This document looks at defining a strategy for extending how tokens are cached internally by MSAL, allowing app developers to cache multiple tokens by a custom key.
+
+## Functional Requirements
+
+- Allow higher level SDKs to extend the default cache key semantics that all MSALs come with.
+- Non-breaking changes
+- Implementations MUST be consistent across confidential client MSALs
+- `AcquireTokenForClient` (client_credentials) MUST support this mechanism. Other confidential client flows (web app, web api, ROPC) SHOULD support it.
+
+## Non-Functional Requirements
+
+- Performance of cache look-up operations does not degrade.
+- Forward compatibility strategy - older MSALs must continue to work on the with the same cache as the new ones, to support upgrade scenarios. If this is not possible, an intermediate version of MSAL must be released that supports this.
```

Comment:
I think I understand what is being asked here, but the wording needs to be fixed so that we are 100% sure.